src/analyze/expr.rs

```diff
@@ -911,12 +911,15 @@ pub(super) fn literal(literal: Literal, location: Location) -> Expr {
 }
 
 fn pointer_promote(left: &mut Expr, right: &mut Expr) -> bool {
+    let is_convertible_to_any_pointer = |expr: &Expr| {
+        expr.ctype.is_void_pointer() || expr.ctype.is_char_pointer() || expr.is_null()
+    };
     if left.ctype == right.ctype {
         true
-    } else if left.ctype.is_void_pointer() || left.ctype.is_char_pointer() || left.is_null() {
+    } else if is_convertible_to_any_pointer(left) && right.ctype.is_pointer() {
         left.ctype = right.ctype.clone();
         true
-    } else if right.ctype.is_void_pointer() || right.ctype.is_char_pointer() || right.is_null() {
+    } else if is_convertible_to_any_pointer(right) && left.ctype.is_pointer() {
```

Comment:
I don't know what these do. Can I get a link to the relevant spec section?

Reply:
http://port70.net/~nsz/c/c11/n1570.html#6.5.15

Reply:
I also added a comment at the start of the function.

Reply:
I feel like this should have been a separate pull request but oh well.

Reply:
Originally I had `#define va_arg(ap, type) (0 ? "va_arg is not currently supported by saltwater" : 0.0)`. But it turns out that I don't give _any_ sort of reasonable errors from macro expansion, so it didn't really matter.

```
stdarg_example.c:4:5 error: invalid program: incompatible types in ternary expression: 'char *' cannot be converted to 'double'
    va_start(args, n);
    ^^^^^^^^
```

Reply:
I improved the error messages a bit in the latest commit:

```c
stddef_example.c:5:21 error: invalid syntax: expected ')', got 'not_currently_supported'
    printf("%ul\n", offsetof(struct s { int i; }, i));
                    ^^^^^^^^
```

Still not great but better than before.